packages/core/src/internal/base/button.base.ts

```diff
@@ -79,15 +74,20 @@ export class CdsBaseButton extends LitElement {
     }, 300);
   }
 
+  private setupNativeButtonBehavior(readonly: boolean) {
+    if (readonly) {
+      this.removeEventListener('click', this.triggerNativeButtonBehavior);
+      this.removeEventListener('keydown', this.emulateKeyBoardEventBehavior);
+    } else {
+      this.addEventListener('click', this.triggerNativeButtonBehavior);
+      this.addEventListener('keydown', this.emulateKeyBoardEventBehavior);
+    }
```

Comment:
Did we add any unit tests for all this? Or did we modify existing ones?

Reply:
These were already present and applied always, but now conditionally applying to buttons that are not readonly.

Looks like we never created a spec file for base button? I can create some tests for the base button in general and include coverage for this part.

Reply:
I think we were testing through `button.spec.ts` to avoid duplication. @coryrylan , plz correct me if i'm wrong on that.

It may be enough to verify this is still adequately tested through those unit tests. Verify code coverage, etc.

Reply:
Added unit test to check add / remove event handlers on `readonly` updates.